sig-apps/Charter.md

```diff
@@ -0,0 +1,141 @@
+  - *MUST* show sustained contributions to at least one subproject or to the SIG
+  - *SHOULD* hold some documented role or responsibility in the SIG and / or at least one subproject
+    (e.g. reviewer, approver, etc)
+  - *MAY* build new functionality for subprojects
+  - *MAY* participate in decision making for the subprojects they hold roles in
+  - Includes all reviewers and approvers in [OWNERS] files for subprojects
+
+## Organizational management
+
+- SIG meets weekly on zoom with agenda in meeting notes
+  - *SHOULD* be facilitated by chairs unless delegated to specific Members
+- SIG overview and deep-dive sessions organized for Kubecon
+  - *SHOULD* be organized by chairs unless delegated to specific Members
+
+- Contributing instructions defined in the SIG CONTRIBUTING.md
+
+### Project management
+
+#### Subproject creation
+
+Subprojects may be created by [KEP] proposal and accepted by [lazy-consensus] with fallback on majority vote of
+SIG Chairs.  The result *SHOULD* be supported by the majority of SIG members.
+  - KEP *MUST* establish subproject owners
+  - [sigs.yaml] *MUST* be updated to include subproject information and [OWNERS] files with subproject owners
+  - Where subprojects processes differ from the SIG governance, they must document how
+    - e.g. if subprojects release separately - they must document how release and planning is performed
+
+Subprojects must define how releases are performed and milestones are set.  Example:
+
+> - Release milestones
+>   - Follows the kubernetes/kubernetes release milestones and schedule
+>   - Priorities for upcoming release are discussed during the SIG meeting following the preceding release and
+>     shared through a PR.  Priorities are finalized before feature freeze.
+> - Code and artifacts are published as part of the kubernetes/kubernetes release
+
+#### Subproject retirement
+
+Subprojects may be retired, where they are archived to the GitHub kubernetes-retired organization, when they are
+no longer supported cased on the following criteria.
+
+- A subproject is no longer supported when there are no active owners with activity on the project for the following time:
+  - Subprojects with no known users can be retired after being unsupported for > 3 months
+  - Subprojects with known users may be retired after providing at least 6 months notification of retirement
+- Use [lazy-consensus] amongst chairs with fallback on majority vote to decide to retire.  This *SHOULD* be
+  supported by a majority of SIG Members.
+
+### Technical processes
+
+Subprojects of the SIG *MUST* use the following processes unless explicitly following alternatives
+they have defined.
+
+- Proposing and making decisions
+  - Proposals against sub-projects, not part of core Kubernetes, will have issues filed against their repositories
+  - When issues are used for Proposals sub-project will have their own decision making process
```

Comment:
Subprojects that are part of SIG Apps can have their own governance outside of SIG Apps? What is value is SIG Apps providing here?

Reply:
This is a great question. It comes back to two things and intent. KEPs and the core/non-core subprojects. KEPs, the process owned by SIG Architecture, is primarily targeted at core but non-core projects are encouraged to use them as well. In the move for core kubernetes to become stable "boring" infrastructure people can rely on there is a move to well justified changes that are thought out and communicated in detail. This is slow and that's good for core and the stability of the project.

Some subprojects need to move fast. For example the application crd/controller. They need to iterate and experiment. They aren't part of core but instead a sig owned repo. KEPs don't always fit well here. This is attempting to give subprojects like this some room to move fast while experimenting.

The application crd/controller makes sense to be owned by SIG Apps because SIG Apps members are collaborating on an interoperability issue that affects numerous open source projects and vendor products. It's a place we can come together on that issue.

Does that help explain it?

Reply:
If non-core subprojects need to move fast and are unwilling to go through the KEP process, do they belong under SIG governance? 

To me, this seems like a backdoor to bypassing official processes for official projects.

My proposition:

These developers are free to experiment and move fast on non-core projects on their own time, but it must not be an official sponsored SIG subproject or take up SIG resources (repos, meetings, upstream CI, etc.)

Reply:
I agree with @mattfarina.

The value of having every subproject owned by some SIG is visibility, accountability, and consistency.

Again, let's table the topic of "non-core" subprojects here.